controllers/class.tapestry-controller.php

```diff
@@ -0,0 +1,337 @@
+<?php
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-errors.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-helpers.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-user-roles.php";
+require_once dirname(__FILE__) . "/../utilities/class.tapestry-node-permissions.php";
+require_once dirname(__FILE__) . "/../interfaces/interface.tapestry-controller.php";
+
+/**
+ * Add/update/retrieve a Tapestry
+ * 
+ */
+class TapestryController implements ITapestryController
+{
+    private $postId;
+
+    /**
+     * Constructor
+     * 
+     * @param   Number  $postId post ID
+     * 
+     * @return  NULL
+     */
+    public function __construct($postId = 0)
+    {
+        $this->postId = (int) $postId;
+    }
+
+    /**
+     * Add A Tapestry
+     * 
+     * @param   Object  $tapestry   Tapestry
+     * 
+     * @return  Object  $tapestry
+     */
+    public function save($tapestry)
```

Comment:
Please create a new method called `set($tapestry)` for setting the tapestry content. You could potentially move the validation in there as well.